go/worker/keymanager/watcher.go

```diff
@@ -303,26 +325,48 @@ func (w *rtNodeWatcher) watch(ctx context.Context) {
 					)
 					return
 				}
+				clear(computeNodes)
 
 				for _, cm := range cms {
 					if cm.Kind != scheduler.KindComputeExecutor {
 						continue
 					}
 
 					for _, member := range cm.Members {
-						_, _ = watcher.WatchNode(ctx, member.PublicKey)
+						nd, err := w.consensus.Registry().GetNode(ctx, &registry.IDQuery{
+							ID:     member.PublicKey,
+							Height: consensus.HeightLatest,
+						})
+						if err != nil {
+							w.logger.Error("failed to fetch node descriptor for committee member",
+								"err", err,
+								"member", member.PublicKey,
+							)
+							continue
+						}
+						computeNodes[nd.ID] = nd
 					}
 				}
 			}()
-		case nu := <-ch:
-			if nu.Reset {
-				// Ignore reset events to avoid clearing the access list before setting a new one.
-				// This is safe because a reset event is always followed by a freeze event after the
-				// nodes have been set (even if the new set is empty).
-				continue
+		case ne := <-nodeCh:
+			if !ne.IsRegistration {
```

Comment:
Maybe use a `switch` for this and add comments for both cases (e.g. registration vs. expiry).